### PR TITLE
ctb: Fix Holocene upgrade script

### PR DESCRIPTION
**Pull request by sebastianst**
**Description**

A couple of fixes to the Holocene upgrade script.

* bumped the contracts release to use to `rc.2`.
* removed `''` from `sed -i ''` invocations - this is a MacOS quirk, but doesn't work in a Linux Docker env.
* deployment script is only run if deployments aren't present yet.
* `deploy-config.json` was made a parameter to `just run`.
* output folder is an optional second parameter to  `just run`. Defaults to `output/`.
* scripts are copied from the checked out repo, not the contracts revision of the monorepo, allowing to use newer scripts (and possibly modify them locally).
* improved documentation in README.

### CODE REVIEW

File: packages/contracts-bedrock/scripts/upgrades/holocene/scripts/main.sh
Comment (clabby) [inline packages/contracts-bedrock/scripts/upgrades/holocene/scripts/main.sh]:
```suggestion
```

File: packages/contracts-bedrock/scripts/upgrades/holocene/scripts/main.sh
Comment (clabby) [inline packages/contracts-bedrock/scripts/upgrades/holocene/scripts/main.sh]:
```suggestion
```

File: packages/contracts-bedrock/scripts/upgrades/holocene/.env.example
Comment (clabby) [inline packages/contracts-bedrock/scripts/upgrades/holocene/.env.example:1]: We can just delete these, yeah?

**Review by clabby — APPROVED** [review]

File: packages/contracts-bedrock/scripts/upgrades/holocene/.env.example
Comment (sebastianst) [reply packages/contracts-bedrock/scripts/upgrades/holocene/.env.example:1]: @clabby what do you want to delete? The `.env.example`? I think we should keep it 😅

Comment (sebastianst) [reply packages/contracts-bedrock/scripts/upgrades/holocene/.env.example:1]: Or do you mean the commented out vars?

Comment (sebastianst) [reply packages/contracts-bedrock/scripts/upgrades/holocene/.env.example:1]: deleted

Comment (mds1) [inline packages/contracts-bedrock/scripts/upgrades/holocene/.env.example:1]: We want the deploy/upgrade script to part of the release, so we need to either:
- Change the target branch to `proposal/op-contracts/v1.8.0`, then cut a new `op-contracts/v1.8.0-rc.3` release after merge, and merge that branch back into develop later
- Merge this, and cherry pick the commit onto the above branch, then cut a new `op-contracts/v1.8.0-rc.3` release after merge

Comment (clabby) [reply packages/contracts-bedrock/scripts/upgrades/holocene/.env.example:1]: Let's do #2. We did the same for rc.2

**Comment by mslipper** [comment]
The issue breaking the merge queue here has been fixed - should be good to merge once the above convo is resolved.

File: packages/contracts-bedrock/scripts/upgrades/holocene/.env.example
Comment (geoknee) [reply packages/contracts-bedrock/scripts/upgrades/holocene/.env.example:1]: I completed these steps, new release is here https://github.com/ethereum-optimism/optimism/releases/tag/op-contracts%2Fv1.8.0-rc.3

**Pull request merged**
